### ISSUE TITLE
output in version 1.5 still uses deprecated features of the CycloneDX schema and should (TMHO) upgrade to use the latest/non-deprecated attributes/schema

### ISSUE THREAD

**Issue by tchinchow** [CycloneDX/cyclonedx-maven-plugin#487]
I recently ran into an issue while trying to merge multiple CycloneDX SBOMs in version 1.5 into one single unified SBOM and though that the error was due to #469.
I waited until #469 was release in 2.8.0 but realized that there may be another problem with the output file format.

Indeed, even when configured to produce CycloneDX v1.5 files, the output still uses deprecated structures for the "tools" section.
_(Please see https://cyclonedx.org/docs/1.5/json/#tab-pane_metadata_tools_oneOf_i0 for reference)_

Here is what I get:
```
{
  ...
  "metadata" : {
    "tools" : [
      {
        "vendor" : "OWASP Foundation",
         ...
      }
    ],
...
```

Here is what the other tools produce _(and what the standard suggests)_:
```
{
  ...
  "metadata" : {
    "tools" : {
      "components": [
        {
           ...
        }
      ],
      "services": [
      ]
    },
...

Do you think you could consider upgrading the v1.5 output file format to use the [recommended schema](https://cyclonedx.org/docs/1.5/json/#tab-pane_metadata_tools_oneOf_i0) instead of the [deprecated structure](https://cyclonedx.org/docs/1.5/json/#tab-pane_metadata_tools_oneOf_i1) ? 
```

Many thanks in advance for your time.

EDIT: Looking at the code, I realize this may actually come from `cyclonedx-core-java` and I submitted an issue there as well  https://github.com/CycloneDX/cyclonedx-core-java/issues/395

**Comment by jkowalleck** [comment]
> [...] should (TMHO) upgrade to use the latest/non-deprecated attributes/schema 

There is no need to do change. The deprecated structures will not go away for a while, unless CycloneDX v2 is being released. 

The new structures might enable new features. The only reason to switch from "old" to "new" structures should be, if you are going to actually use the features the new structures introduced. 

CC @CycloneDX/core-team

**Comment by tchinchow** [comment]
Thanx for your answer.

I'm "kind of" going to use the new structure because I'm merging a Maven SBOM _(back end)_ with another SBOM _(for the front end)_ generated with a tool called Syft which uses the new format.

So Yes because merging the files into one _is_ using the files... But I reckon that's far fetched and somehow opinionated ;) .

Currently the two files are incompatible while being in the same format and same version: Only generated by two different tools. Thinking about how this is even possible casts a pretty dark shadow on CycloneDx itself and the promise of interoperability but I'll be constructive here : where do we go from here ?

I can work around the problem and replace the generated deprecated part because it is pretty thin.
The question then would be "what am I going to hardcode in that (metadata/tools) section ?" ...
I'm asking because the part in question describes is the `cyclonedx-maven-plugin` itself !

If this isn't going to be patched, maybe people would be interested in knowing how you would describe `cyclonedx-maven-plugin` with the new syntax:
- component or service ?
- if "component", then what type of component ?

Can someone draw a simple sketch of how `cyclonedx-maven-plugin` would look like in the new format or at least provide directions for the two above questions ?

**Comment by jkowalleck** [comment]
re: https://github.com/CycloneDX/cyclonedx-maven-plugin/issues/487#issuecomment-2069363661 


So you decided to consume and merge(transform) multiple CycloneDX documents? 
Well, I guess it is the task of the merge tool, to assure a valid outcome, right? This means , the merge tool must take care of possible conversions - for example, streamlining `$.metadata.tools.tool` and `$.metadata.tools.component`.

----

example conversion (json with comemnts)

from:
```json5
{ // this is metadata
  "tools: [
    { // this is a tool
     "vendor": "my-vendor",
     "name": "my-name",
     "version": "1.33.7",
     "hashes": [ /* my-hashes */ ]
     "externalReferences": [ /* my-extRefs*/ ]
    }
] }
```

to:
```json5
{ // this is metadata
  "tools: {
    "components": [
      { // this is the "tool" from above, as a compoennt representation
        "type": "application"
        "group": "my-vendor",
        "name": "my-name",
        "version": "1.33.7",
        "hashes": [ /* my-hashes */ ]
        "externalReferences": [ /* my-extRefs*/ ]
      }
    ]
  }
}
```

**Comment by hboutemy** [comment]
I'm reopening the issue: the discussion on merge tools being able to manage and transform is a good one, but cyclonedx-maven-plugin should generate non-obsolete fields, which is not really a problem given there are transformation rules

I want to implement this soon, and do a 2.8.1 release for CycloneDX 1.5 before going to 2.9 for CycloneDX 1.6

**Comment by hboutemy** [comment]
@jkowalleck @tchinchow please review #517

**Comment by tchinchow** [comment]
> @jkowalleck @tchinchow please review #517

Sorry for the late answer.
I saw that #517 has been merged in version 2.8.1 ... So I have tried version 2.8.1 instead and can confirm it works like a charm... I no longer have to patch the produced output manually.

Thanks a lot for this !
